src/optimizer_reader.c

```diff
@@ -233,7 +233,8 @@ IndexIterator *NewOptimizerIterator(QOptimizer *qOpt, IndexIterator *root, Itera
   oi->lastLimitEstimate = qOpt->nf->limit =
     QOptimizer_EstimateLimit(oi->numDocs, oi->childEstimate, qOpt->limit);
 
-  const FieldSpec *field = IndexSpec_GetField(qOpt->sctx->spec, qOpt->nf->fieldName, strlen(qOpt->nf->fieldName));
+  // TODO: need to have the field name inside the iterator tree to be a hidden name
```

Comment:
Let's have this as a Jira ticket, not a comment in the code

Reply:
I think Guy's changes might make that redundant.
Removed the comment altogether.